db/db_impl_secondary.cc

```diff
@@ -52,12 +58,167 @@ Status DBImplSecondary::Recover(
     default_cf_internal_stats_ = default_cf_handle_->cfd()->internal_stats();
     single_column_family_mode_ =
         versions_->GetColumnFamilySet()->NumberOfColumnFamilies() == 1;
+
+    // Recover from all newer log files than the ones named in the
+    // descriptor.
+    std::vector<std::string> filenames;
+    s = env_->GetChildren(immutable_db_options_.wal_dir, &filenames);
+    if (s.IsNotFound()) {
+      return Status::InvalidArgument("Failed to open wal_dir",
+                                     immutable_db_options_.wal_dir);
+    } else if (!s.ok()) {
+      return s;
+    }
+
+    std::vector<uint64_t> logs;
+    for (size_t i = 0; i < filenames.size(); i++) {
+      uint64_t number;
+      FileType type;
+      if (ParseFileName(filenames[i], &number, &type) && type == kLogFile) {
+        logs.push_back(number);
+      }
+    }
+
+    if (!logs.empty()) {
+      // Recover in the order in which the logs were generated
+      std::sort(logs.begin(), logs.end());
```

Comment:
If `log_readers_` is not empty, then we can remove the log numbers that are smaller than the smallest element in `log_readers_`.

Reply:
in the current implementation, `log_readers_` is always empty here, so the filtering is no-op right now. Shall we add the filtering at a later time when `log_readers_` lifecycle changes?

Reply:
Oh, I was thinking about next step where we read and apply the WAL dynamically. Fixing one line here should work for that case. Your call.

Reply:
added the filtering, PTAL